cli/cmd/offsets.go

```diff
@@ -0,0 +1,235 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/odigos-io/odigos/api/k8sconsts"
+	cmdcontext "github.com/odigos-io/odigos/cli/pkg/cmd_context"
+	"github.com/odigos-io/odigos/cli/pkg/kube"
+	"github.com/odigos-io/odigos/common/consts"
+)
+
+var (
+	offsetFileMountPath = "/offsets"
+	useDefault          bool
+)
+
+var offsetsCmd = &cobra.Command{
+	Use:   "update-offsets",
+	Short: "Update Odiglet to use the latest available Go instrumentation offsets",
+	Long: `This command pulls the latest available Go struct and field offsets information from Odigos.
+It stores this data in a ConfigMap in the Odigos Namespace and updates the Odiglet DaemonSet to mount it.
+
+Use this command when instrumenting apps that depend on very new dependencies that aren't currently supported
+with the previous release of Odigos.
+
+Note that updating offsets does not guarantee instrumentation for libraries with significant changes that
+require an update to Odigos. See docs for more info: https://docs.odigos.io/instrumentations/golang/ebpf#about-go-offsets
+`,
+	Example: `
+# Pull the latest offsets and restart Odiglet
+odigos update-offsets
+
+# Revert to using the default offsets data shipped with Odigos
+odigos update-offsets --default
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+		client := cmdcontext.KubeClientFromContextOrExit(ctx)
+		ns := cmd.Flag("namespace").Value.String()
+
+		resp, err := http.Get(consts.GoOffsetsPublicURL)
+		if err != nil {
+			fmt.Println(fmt.Sprintf("\033[31mERROR\033[0m Cannot get latest offsets: %s", err))
```

Comment:
can we check here if the error might be no internet access and add a note about it? so users are aware this requires internet and cannot work offline.